### PR TITLE
git-import: fix issue when first commit is a merge

### PR DESCRIPTION
**Pull request by aplanas**

### CODE REVIEW

File: git-importer.py [hunk 1]
```diff
@@ -374,6 +368,21 @@ def merge(
...
+        except:
+            return None
+
+    def clean(self):
+        for path, _ in self.repo.status().items():
+            logging.debug(f"Cleaning {path}")
+            try:
+                (self.path / path).unlink()
+                self.repo.index.remove(path)
+            except Exception:
```
Comment (coolo) [inline git-importer.py:383]: I take a PIP and throw it at you :)

Comment (aplanas) [reply git-importer.py]: Yeah ... Added a logging.warning, and continue

**Comment by coolo** [comment]
When I run your branch on Colorfull, I get
image

I.e. there are no "Accepted Request" commits in Factory.

**Comment by coolo** [comment]
What is the tool creating the ascii tree?

I get that 437901 does not create a merge commit, but I guess it should be an empty commit still.

**Comment by aplanas** [comment]
> What is the tool creating the ascii tree?

Not mine:

```
git log --graph --abbrev-commit --decorate --format=format:'%C(bold blue)%h%C(reset) - %C(bold green)(%ar)%C(reset) %C(white)%<(50,trunc)%s%C(reset) %C(dim white)- %an%C(reset)%C(bold yellow)%d%C(reset)' --all
```

You can remove `%<(50,trunc)` to have the full message.

> I get that 437901 does not create a merge commit, but I guess it should be an empty commit still.

Uhmm ... can be, yes.

I was trying to map when "devel" it is created first, and we do a SR to Factory to create it in the distribution.

In `git` I was hopping that this was the right thing to do: start in the `devel` branch, do a series of commits there, and branch `factory` from the first commit that make it appear.

**Comment by coolo** [comment]
> In git I was hopping that this was the right thing to do: start in the devel branch, do a series of commits there, and branch factory from the first commit that make it appear.

Yes, but still the action to accept it into Factory should be represented IMO

**Comment by coolo** [comment]
(and gitk --all doesn't show the full tree for whatever reason, `git log --graph` does )

**Comment by aplanas** [comment]
> (and gitk --all doesn't show the full tree for whatever reason, `git log --graph` does )

This is what I see with `gitk --all`, from an export that I just did. I checked that there is no any pending rebase or local change in the code:

gitk

**Comment by aplanas** [comment]
> * XXXXXX - (6 years ago) - Accepted into Factory - OBS User lnussel_suse

I see. This would be an empty commit with `parent = 5917b9c`

**Comment by coolo** [comment]
I'm merging this to avoid further conflicts and I'll add a TODO in the code - hopefully you find time soonish to enhance this

File: git-importer.py [hunk 2]
```diff
@@ -917,6 +927,24 @@ def import_all_revisions(self):
...
+    def import_new_revision_with_request(self, revision, request):
+        """Create a new branch as a result of a merge"""
+
+        submitted_revision = self.history.find_revision(
+            request.source, request.revisionid, revision.time
+        )
+        if not submitted_revision:
+            logging.warning(f"Request {request} does not connect to a known revision")
+            return False
+        assert submitted_revision.commit is not None
```
Comment (coolo) [inline git-importer.py:939]: this assertion hits 60 packages now. 

```
(venv) importer@git-importer:~/logs> cat FAIL.premake4
CACHE_DIR /home/importer/.cache/openSUSE-release-tools/request/main
INFO:root:Importing [Rev openSUSE:Factory/1 Md5 9b55eae0a1f723d7d9f7411f87b271cc 2011-04-29 10:50:37 saschpe 68121] to factory
Traceback (most recent call last):
  File "/home/importer/openSUSE-release-tools/git-importer.py", line 1192, in <module>
    main()
  File "/home/importer/openSUSE-release-tools/git-importer.py", line 1188, in main
    importer.import_all_revisions()
  File "/home/importer/openSUSE-release-tools/git-importer.py", line 940, in import_all_revisions
    self.import_revision(revision)
  File "/home/importer/openSUSE-release-tools/git-importer.py", line 1087, in import_revision
    self.import_new_revision_with_request(revision, request)
  File "/home/importer/openSUSE-release-tools/git-importer.py", line 951, in import_new_revision_with_request
    assert submitted_revision.commit is not None
AssertionError

Comment (aplanas) [reply git-importer.py:939]: seems that is again an issue with the ordering:

```
DEBUG:root:Selected import order for premake4
DEBUG:root:Rev openSUSE:Factory/1 Md5 9b55eae0a1f723d7d9f7411f87b271cc 2011-04-29 12:50:37 saschpe 68121
DEBUG:root:Rev devel:tools:building/1 Md5 ebc0070db1cced22695e7659707a65b5 2011-04-29 12:50:39 saschpe 68121
```

Comment (aplanas) [reply git-importer.py:939]: We should build the graph ASAP and drop this mechanism.

Comment (aplanas) [reply git-importer.py:939]: https://github.com/openSUSE/openSUSE-release-tools/pull/2862

**Pull request merged**